can/interfaces/usb_can_analyzer/usb_can_analyzer.py

```diff
@@ -0,0 +1,276 @@
+                      ``is_error_frame`` will be ignored.
+
+            .. note:: If the timestamp is a float value it will be converted
+                      to an integer.
+
+        :param timeout:
+            This parameter will be ignored. The timeout value of the channel is
+            used instead.
+
+        """
+        try:
+            timestamp = struct.pack('<I', int(msg.timestamp * 1000))
+        except struct.error:
+            raise ValueError('Timestamp is out of range')
+        try:
+            a_id = struct.pack('<I', msg.arbitration_id)
+        except struct.error:
+            raise ValueError('Arbitration Id is out of range')
+        byte_msg = bytearray()
+        byte_msg.append(0xAA)
+        for i in range(0, 4):
+            byte_msg.append(timestamp[i])
+        byte_msg.append(msg.dlc)
+        for i in range(0, 4):
+            byte_msg.append(a_id[i])
+        for i in range(0, msg.dlc):
+            byte_msg.append(msg.data[i])
+        byte_msg.append(0xBB)
+        self.ser.write(byte_msg)
+
+    def _recv_internal(self, timeout):
+        """
+        Read a message from the serial device.
+
+        :param timeout:
+
+            .. warning::
+                This parameter will be ignored. The timeout value of the channel is used.
+
+        :returns:
+            Received message and False (because not filtering as taken place).
+
+            .. warning::
+                Flags like is_extended_id, is_remote_frame and is_error_frame
+                will not be set over this function, the flags in the return
+                message are the default values.
+
+        :rtype:
+            can.Message, bool
+        """
+        try:
+            # ser.read can return an empty string
+            # or raise a SerialException
+            rx_byte_1 = self.ser.read()
+            
+        except serial.SerialException:
+            return None, False
+
+        if rx_byte_1 and  ord(rx_byte_1) == 0xAA:
+            rx_byte_2 = ord(self.ser.read())
+            time_stamp = time()
+            if rx_byte_2 == 0x55:
+                rx_msg_type = self.ser.read()
+
+            else:
+                length = int(rx_byte_2 & 0x0F)
+                is_extended = bool(rx_byte_2 & 0x20)
+                is_data = bool(rx_byte_2 & 10)
+
+                if is_extended:
+                    s_3_4_5_6 = bytearray(self.ser.read(4))
+                    arb_id = (struct.unpack('<I', s_3_4_5_6))[0]
+
+                else:
+                    s_3_4 = bytearray(self.ser.read(2))
+                    arb_id = (struct.unpack('<H', s_3_4))[0]
+
+                data = bytearray(self.ser.read(length))
+                end_packet = ord(self.ser.read())
+
+                if end_packet == 0x55:
+                    msg = Message(timestamp=time_stamp,
+                                  arbitration_id=arb_id,
+                                  extended_id=is_extended,
+                                  data=data)
```

Comment:
Could also add `is_remote_frame=not is_data`.

Reply:
should be done, 
its set when remote: ie 1='is_remote', changed var name and fixed mask to be 0x10.

Reply:
Oh and also add the dlc argument.

Reply:
The DLC is being set in line 252, right?